torch/csrc/jit/function.h

```diff
@@ -27,6 +27,12 @@ struct TORCH_API Function {
     get_executor().run(stack);
   }
 
+  Function(const Function& other)
+      : name_(other.name_),
+        graph_(other.graph_->copy()),
+        optimize_(other.optimize_),
+        function_creator_(other.function_creator_) {}
```

Comment:
We should fail if function_creator_ is set. Copying the function won't work correctly. I am also wary of making this public, because now people can copy Function objects without error. That seems bad. They are not quite public objects, but they behave very differently than our actual public objects like Module IValue, Tensor, which can be copied but do so by reference.